src/Navigation/Navigate.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * @Author: Erik Slovak <erik.slovak@technologystudio.sk>
+ * @Date: 2022-11-02T14:11:60+01:00
+ * @Copyright: Technology Studio
+**/
+
+import { last } from '@txo/functional'
+import { Log } from '@txo/log'
+
+import {
+  conditionalNavigationManager,
+} from '../Api/ConditionalNavigationManager'
+import type {
+  OnActionAttributes,
+  ResolveConditionsResult,
+  WithConditionalNavigationState,
+} from '../Model/Types'
+import {
+  getActiveLeafRoute,
+  getExistingRouteByRouteName,
+  getRoutePathFromAction,
+  getScreenNavigationConditions,
+} from '../Api/NavigationUtils'
+
+const log = new Log('txo.react-conditional-navigation.Navigation.Navigate')
+
+const VOID = 'void'
+
+export const onNavigateAction = ({
+  action,
+  getContext,
+  getRootState,
+  nextOnAction,
+  originalOnAction,
+  restArgs,
+  screenConditionConfigMap,
+  setState,
+}: OnActionAttributes): boolean => {
+  const {
+    payload,
+    flow,
+    reset,
+    skipConditionalNavigation,
+  } = action
+  const navigationState = getRootState?.()
+
+  const nextRoutePath = getRoutePathFromAction(action) ?? []
+  const leafRouteName = last(nextRoutePath)
+  log.debug('NAVIGATE', { action, navigationState })
+  if (!skipConditionalNavigation) {
+    if (navigationState) {
+      let resolveConditionsResult: ResolveConditionsResult | undefined
+      for (const routeName of nextRoutePath) {
+        const screenConditions = getScreenNavigationConditions(screenConditionConfigMap[routeName])
+        if (screenConditions && screenConditions.length > 0) {
+          resolveConditionsResult = conditionalNavigationManager.resolveConditions(screenConditions, action, navigationState, getContext) ?? resolveConditionsResult
+        }
+      }
+      log.debug('N: RESOLVE CONDITIONS RESULT', { resolveConditionsResult, action, _conditionToResolveCondition: conditionalNavigationManager._conditionToResolveCondition, screenConditionConfigMap })
+      if (resolveConditionsResult) {
+        const activeLeafRoute = getActiveLeafRoute(navigationState)
+        activeLeafRoute.conditionalNavigation = resolveConditionsResult.conditionalNavigationState
+        return nextOnAction(resolveConditionsResult.navigationAction, ...restArgs)
+      }
```

Comment:
this should be processed for the first non nillable resolveConditionsResult and not for the last, you can move the whole section directly after line 56.